### DISCUSSION TITLE
Start using GT4SD

### DISCUSSION THREAD

**Discussion by PT2727** [GT4SD/gt4sd-core#233] [Q&A]
Dear community,

I just recently started trying to use GT4SD, especially I am interested in knowing more about the discovery-demo example. However, after installing gt4sd, I am unable to run the commands. 

I attached the error that I received after running 
>> _from gt4sd.properties import PropertyPredictorRegistry_

[gt4sd-error.docx](https://github.com/GT4SD/gt4sd-core/files/13460241/gt4sd-error.docx)

Could anyone help with solving this problem?
I am very much a beginner with correctly installing things on a linux computer.
Thank you in advance!

**Comment by drugilsberg** [comment]
Hi @PT2727, looks like a CUDA related error.
Are you running this code on GPU? If yes, it looks like you compiled gt4sd for CPU you should reinstall the library using the dedicated environment file:

```console
conda env create -f conda_gpu.yml
```

**Reply by PT2727** [reply]
Hi @drugilsberg, 
I am running it on a CPU cluster and installed it using conda.yml (not conda_gpu.yml)

I do have access to a gpu-cluster (which I have never used before), in case I should run it on there.
I did indeed compile it for cpu.

**Comment by drugilsberg** [comment]
It should be fine, there was a misleading warning about not having a compiled version for GPU, that's why I asked.

I never observed this error, but to me it seems to be an installation issue. It looks like you have a dynamic library with a missing symbol:

```console
ImportError: /home/pcpt3/.conda/envs/gt4sd/lib/python3.8/site-packages/fast_transformers/causal_product/causal_product_cpu.cpython-38-x86_64-linux-gnu.so: undefined symbol: 
_ZTIN3c1021AutogradMetaInterfaceE
```

I would recommend reinstalling the environment from scratch saving the output to test to check whether there are some error messages appearing during the installation.

Something like this should do the trick:

```console
# first remove the existing env
rm -rf /home/pcpt3/.conda/envs/gt4sd/
# reinstall saving the build output to file
conda env create -f conda.yml > gt4sd-build.log
```

**Reply by PT2727** [reply]
Good idea!

Before I do this, I actually saved the ouptut I got when I installed it a few weeks ago.
Maybe this helps already

[gt4sd-installation.docx](https://github.com/GT4SD/gt4sd-core/files/13460917/gt4sd-installation.docx)

**Reply by drugilsberg** [reply]
Indeed i don't see any apparent error. As the error is coming from a dependency called fast_transforners when loading a dynamic library it might be worth reinstalling the dependency.

If this does not help I would recommend looking for the error in Relation with fast_transforners.

**Reply by PT2727** [reply]
Okay thank you!
How do I reinstall the dependency (I am quite new with this?.
Do I completely recreate the environment and pip install gt4sd, or are their easier ways to just tackle fast_transformers?

**Reply by drugilsberg** [reply]
It's sufficient to uninstall it and reinstall it in the activated conda env:

```console
conda activate gt4sd
pip uninstall pytorch-fast-transformers
pip install pytorch-fast-transformers==0.4.0
```

**Reply by PT2727** [reply]
Thank you for explaining! I tried this and it is giving the same error. Ill try asking this question on the fast_transformers github page.

**Reply by drugilsberg** [reply]
Sounds good, unfortunately I'm not able to reproduce the error at the moment.
